Activities/Cryptography/UiPath.Cryptography/CryptographyHelper.cs

```diff
@@ -260,5 +315,145 @@ private static int[] GetLegalKeySizes(SymmetricAlgorithm algorithm)
 
             return keySizes.ToArray();
         }
+
+#if NET5_0
+        private static byte[] EncryptAesGcmNet5(byte[] inputBytes, byte[] key, out byte[] result)
```

Comment:
There's a lot of overlap between this method and the 461 one.

Same for the decryption counterparts.

Can we extract the common parts into dedicated methods and reuse them?

Or, since it seems that only the part inside the `using`s is different, maybe create some helper methods around those and have a single set of Encrypt/Decrypt methods, which call the appropriate helper methods inside them via `#ifdef` ?

Something like

```
private static byte[] EncryptAesGcm(...)
{
// Init

using()
{
#if NET5_0
EncryptAesGcmWith50(...);
#endif

#if NET461
EncryptAesGcmWith461(...);
#endif
}

// Return result
}
```

This also has the advantage of removing a lot of the preprocessing directives elsewhere, since we have a common entry point method for both NET 5.0 and NET 461, and we only need preprocessing inside that method.

Either way, we need to cut down on the code duplication.

Reply:
modified